src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -635,23 +635,23 @@ async Task OnStreamCanceled(InvocationRequest irq)
             return channel;
         }
 
-        private Dictionary<string, object> PackageStreamingParams(ConnectionState connectionState, ref object[] args, out List<string> streamIds)
+        private Dictionary<string, object>? PackageStreamingParams(ConnectionState connectionState, ref object?[] args, out List<string>? streamIds)
         {
-            Dictionary<string, object> readers = null;
+            Dictionary<string, object>? readers = null;
             streamIds = null;
-            var newArgs = new List<object>(args.Length);
+            var newArgs = new List<object?>(args.Length);
 
             for (var i = 0; i < args.Length; i++)
             {
-                if (args[i] != null && ReflectionHelper.IsStreamingType(args[i].GetType()))
+                if (args[i] != null && ReflectionHelper.IsStreamingType(args[i]!.GetType()))
```

Comment:
```suggestion
                if (args[i] is {} arg && ReflectionHelper.IsStreamingType(arg.GetType()))
```
For ultimate @pranavkm points 🐴 

(I think this works)